arrows/qt/widgets/LoggerView.cxx

```diff
@@ -0,0 +1,71 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+#include "LoggerView.h"
+
+#include "ui_LoggerView.h"
+
+namespace kv = kwiver::vital;
+
+namespace kwiver {
+namespace arrows {
+namespace qt {
+
+//-----------------------------------------------------------------------------
+class LoggerViewPrivate
+{
+public:
+
+  Ui::LoggerView UI;
+};
+
+QTE_IMPLEMENT_D_FUNC(LoggerView)
+
+
+//-----------------------------------------------------------------------------
+LoggerView::LoggerView(
+  QWidget* parent, Qt::WindowFlags flags)
+  : QWidget{parent, flags}, d_ptr{new LoggerViewPrivate}
+{
+  QTE_D();
+
+  // Set up UI
+  d->UI.setupUi(this);
+
+  QFont font("Monospace", 10);
+  font.setStyleHint(QFont::TypeWriter);
+  d->UI.loggerText->document()->setDefaultFont(font);
+
+  QObject::connect(this, &LoggerView::logMessage,
+                   this, &LoggerView::appendMessage);
+}
+
+//-----------------------------------------------------------------------------
+LoggerView::~LoggerView()
```

Comment:
Is there a good reason to implement an empty destructor? Not a big deal but if we aren't actually adding anything to it, it might be better to let the compiler generate it.

Reply:
How do you feel about "the code won't compile otherwise" as a reason? :slightly_smiling_face:

Recall that the destructor also destroys member variables, even though those are not written out. One of the member variables is a `QSharedPointer<LoggerViewPrivate>`. The compiler-generated destructor is inline, and will fail to compile in other TUs because a user of `LoggerView` can't destroy that member because it (rightly!) doesn't have a definition of `LoggerViewPrivate`. Thus, the dtor *must* be explicitly instantiated in the `.cxx` where `LoggerViewPrivate`is defined.

That said, `LoggerView::~LoggerView() = default;` would be fine also (I think I've used that at least once), but doesn't bring the usual benefits of an explicitly defaulted dtor. It's also technically *more* bytes (but fewer *lines*).

(Can you work around this by declaring an explicit instantiation of  `QSharedPointer<LoggerViewPrivate>`? Maybe, but I've never bothered.)

Reply:
Yep, that's a good enough reason for me!